capstone/notebooks/K_nearest_neighbour.ipynb

```diff
@@ -0,0 +1,309 @@
+    "    :params class_list: list of relevant image classes\n",
+    "    :params image_names: dictionary of image names for each class\n",
+    "    :params hist_type: name of method for calculating histogram; choices are: (greyscale_histogram, \n",
+    "        color_histogram, complete_histogram)\n",
+    "    :params weight_type: type of weighting to use in KNN model; Uniform or Weighted\n",
+    "    :return: prints out classification accuracy\n",
+    "    \"\"\"    \n",
+    "    # Split into test and training data\n",
+    "    train = {}\n",
+    "    test = {}\n",
+    "    for k, v in image_names.items():\n",
+    "        np.random.shuffle(v)\n",
+    "        train[k] = v[:750]\n",
+    "        test[k] = v[750:]\n",
+    "        \n",
+    "    # Create actual labels and data\n",
+    "    labels = []\n",
+    "    data = []\n",
+    "\n",
+    "    for c in class_list:\n",
+    "        for i in train[c]:\n",
+    "            img = io.imread(os.path.join('../data/food-101/top_classes/', c, '', i))\n",
+    "            if hist_type == 'Greyscale':\n",
+    "                hist, bins = hc.greyscale_histogram(img)\n",
+    "            elif hist_type == 'RGB':\n",
+    "                hist = hc.color_histogram(img)\n",
+    "            elif hist_type == 'Complete':\n",
+    "                hist = hc.complete_histogram(img)\n",
+    "            labels.append(c)\n",
+    "            data.append(hist)\n",
+    "            \n",
+    "    # Initialize model\n",
+    "    if weight_type == 'Uniform':\n",
+    "        neigh = KNeighborsClassifier(n_neighbors=10)\n",
+    "    elif weight_type == 'Weighted':\n",
+    "        neigh = KNeighborsClassifier(n_neighbors=10, weights='distance')\n",
+    "    \n",
+    "    # Fit the model to the data\n",
+    "    neigh.fit(np.array(data), np.array(labels))\n",
+    "    \n",
+    "    # Test the fitted model on the test data\n",
+    "    \n",
+    "    actual = []\n",
+    "    test_data = []\n",
+    "\n",
+    "    for c in class_list:\n",
+    "        for i in test[c]:\n",
+    "            img = io.imread(os.path.join('../data/food-101/top_classes/', c, '', i))\n",
+    "            if hist_type == 'Greyscale':\n",
+    "                hist, bins = hc.greyscale_histogram(img)\n",
+    "            elif hist_type == 'RGB':\n",
+    "                hist = hc.color_histogram(img)\n",
+    "            elif hist_type == 'Complete':\n",
+    "                hist = hc.complete_histogram(img)\n",
+    "            actual.append(c)\n",
+    "            test_data.append(hist)\n",
+    "\n",
+    "    predicted = neigh.predict(np.array(test_data))\n",
+    "    actual = np.array(actual)\n",
+    "    classification_rate = sum(predicted == actual) / len(predicted)\n",
+    "\n",
+    "    print(\"Using {} histogram and {} voting, classification rate is {:.2f}%\".format(hist_type, weight_type, classification_rate*100))"
```

Comment:
Thanks for the comments!
I have committed a new version including these statistics.
It would be good to discuss the interpretation of some of these on Sunday, particular the F1 metric as I am a bit unsure of how to think about it in practice.